otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -100,28 +100,31 @@ caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
 
   struct caml_runtime_events_cursor *cursor =
       caml_stat_alloc_noexc(sizeof(struct caml_runtime_events_cursor));
+  int should_free_events_loc;
```

Comment:
I'd rather see that variable initialized to zero at the declaration point, so that future changes in this routine do not risk using an uninitialized value.

Reply:
Following in that style should `int ret;` get initialised to something since it is being returned at the end of the function?

Reply:
That wouldn't hurt either.

Reply:
I made `ret` uninitialised specifically so the compiler would emit a warning if it isn't assigned before jumping; and `should_free_events_loc` uninitialised because it's to be initialised where `runtime_events_loc` is.

A change that would accidentally use these with the default initialisation is already wrong, so my reasoning is that it's better to have an obviously invalid value that the compiler would yell at us about.

Reply:
This is relying upon the C compiler being always correct in figuring out whether a variable is initialized or not. It turns out they aren't always, with both false positives and false negatives. I wouldn't put too much trust in the compiler.

Reply:
Drive-by comment: do we need a boolean to tell us to free `runtime_events_loc`, or could we just free exactly when the variable is not `NULL`?

Reply:
> Drive-by comment: do we need a boolean to tell us to free runtime_events_loc, or could we just free exactly when the variable is not NULL?

It's not the `NULL` case we're worried about, but the case where it's initialised to `caml_runtime_events_current_location()`, which isn't a copy.

Reply:
I see. I looked out of curiosity, wondering if just making a copy in that case would make things nicer. My quick skim suggests that the function is sprawling and somewhat of a mess, and I think that this is going to bite us again in the future. So I would encourage you to refactor it as you see fit -- for example maybe the system-dependent bits could be factored out into their own helper functions -- to make the code nicer, and not just safer. Your current change is okay but it does not make the code nicer to read.

Reply:
I've refactored `caml_runtime_events_create_cursor` and made `runtime_events_loc` be a new string every time to make reasoning easier.